dlib/bigint/bigint_kernel_2.cpp

```diff
@@ -1528,7 +1528,7 @@ namespace dlib
             const uint32 len = lhs->digits_used + rhs->digits_used;
             for (unsigned long i = 0; i < len; ++i)
             {
-                uint64 num1 = static_cast<uint64>(std::floor(a[i*2].real()+0.5));
+                uint64 num1 = static_cast<uint64>(std::round(a[i*2].real()));
```

Comment:
I used `std::round` instead of `std::lround` here not to lose precision (`long` vs `uint64`)